test/unit-test/cellular_common_utest.c

```diff
@@ -1581,3 +1592,31 @@ void test__Cellular_LibInit_PktHandlerInit_Double_Allocate_Context( void )
     CellularHandle = NULL;
     _Cellular_LibInit( &CellularHandle, &CellularCommInterface, &tokenTable );
 }
+
+/**
+ * @brief Test that null parameter case for _Cellular_RegisterUndefinedRespCallback.
+ */
+void test__Cellular_RegisterUndefinedRespCallback_Null_Parameter( void )
+{
+    CellularContext_t context;
+    CellularPktStatus_t pktStatus = CELLULAR_PKT_STATUS_OK;
+
+    pktStatus = _Cellular_RegisterUndefinedRespCallback( NULL, cellularUndefinedRespCallback );
+    TEST_ASSERT_NOT_EQUAL( CELLULAR_PKT_STATUS_OK, pktStatus );
+
+    /* Unregister undefined response callback. */
+    pktStatus = _Cellular_RegisterUndefinedRespCallback( &context, NULL );
+    TEST_ASSERT_EQUAL( CELLULAR_PKT_STATUS_OK, pktStatus );
+}
+
+/**
+ * @brief Test that happy path case for _Cellular_RegisterUndefinedRespCallback.
+ */
+void test__Cellular_RegisterUndefinedRespCallback_Happy_Path( void )
+{
+    CellularContext_t context;
+    CellularPktStatus_t pktStatus = CELLULAR_PKT_STATUS_OK;
+
+    pktStatus = _Cellular_RegisterUndefinedRespCallback( &context, cellularUndefinedRespCallback );
```

Comment:
Is there a way to ensure that the function actually registers the callback and not just returned CELLULAR_PKT_STATUS_OK?

Reply:
The prototype of _Cellular_RegisterUndefinedRespCallback is update with
```
CellularError_t _Cellular_RegisterUndefinedRespCallback( CellularContext_t * pContext,
                                                         CellularUndefinedRespCallback_t undefinedRespCallback,
                                                         void * pCallbackContext );
```
undefinedRespCallback and pCallbackContext can be NULL to unregister the callback and callback context.
The test case is update with the following situation
* Register the cellular undefined response callback.
* Unregister the cellular undefined response callback.
* Register the cellular undefined response callback without callback context.
* Unregister the cellular undefined response callback without callback context.
The internal filed of CellularContext is also checked in the test case to ensure the function actually registers.